project device/google/crosshatch/
diff --git a/BoardConfigLineage.mk b/BoardConfigLineage.mk
index 70f1f09f..7d60d86f 100644
--- a/BoardConfigLineage.mk
+++ b/BoardConfigLineage.mk
@@ -1,7 +1,7 @@
 # Common board config for crosshatch, blueline
 
-BOARD_AVB_MAKE_VBMETA_IMAGE_ARGS += --set_hashtree_disabled_flag
-BOARD_AVB_MAKE_VBMETA_IMAGE_ARGS += --flags 2
+BOARD_AVB_ALGORITHM := SHA256_RSA2048
+BOARD_AVB_KEY_PATH := vendor/extra/security/verifiedboot.key
 
 BOARD_KERNEL_IMAGE_NAME := Image.lz4
 TARGET_COMPILE_WITH_MSM_KERNEL := true
