project device/google/wahoo/
diff --git a/BoardConfigLineage.mk b/BoardConfigLineage.mk
index 7990e0a..0f8f792 100644
--- a/BoardConfigLineage.mk
+++ b/BoardConfigLineage.mk
@@ -24,5 +24,5 @@ DEVICE_MANIFEST_FILE += device/google/wahoo/lineage_manifest.xml
 DEVICE_FRAMEWORK_COMPATIBILITY_MATRIX_FILE += device/google/wahoo/lineage_compatibility_matrix.xml
 
 # Verified Boot
-BOARD_AVB_MAKE_VBMETA_IMAGE_ARGS += --set_hashtree_disabled_flag
-BOARD_AVB_MAKE_VBMETA_IMAGE_ARGS += --flags 2
+BOARD_AVB_ALGORITHM := SHA256_RSA2048
+BOARD_AVB_KEY_PATH := vendor/extra/security/verifiedboot.key
 